src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Microsoft.Data.Analysis
+{
+    public static class IDataViewExtensions
+    {
+        public static DataFrame ToDataFrame(this IDataView dataView, long maxRows = -1)
```

Comment:
I wonder if we should cap this by default.... it seems dangerous to get all rows by default, could easily OOM the process if you called it on something too big. Having `-1` meaning "give me all", makes sense. I am just questioning if that is the best default.

Reply:
The default of `.Preview()` is 100 rows for the same reasons you mentioned. I think we should cap it at a "safe" value.

Reply:
Good observation. Capping it at 100 to match the behavior of preview makes sense to me.